### PR TITLE
examples/dmabuf-capture: stop using av_init_packet

### PR DESCRIPTION
**Pull request by emersion**
It's deprecated in ffmpeg >= 4.4.

Closes: https://github.com/swaywm/wlroots/issues/2798

### CODE REVIEW

File: examples/dmabuf-capture.c [hunk 1]
```diff
@@ -506,10 +505,10 @@ static void *vid_encode_thread(void *arg) {
 				goto end;
 			}
```
Comment (kennylevinsen) [inline examples/dmabuf-capture.c:506]: If any of the above branches are hit, the AVPacket will be leaked.

Comment (emersion) [reply examples/dmabuf-capture.c:506]: Fixed

File: examples/dmabuf-capture.c [hunk 2]
```diff
@@ -490,10 +491,8 @@ static void *vid_encode_thread(void *arg) {
...
 
 		while (1) {
-			AVPacket pkt;
-			av_init_packet(&pkt);
-
-			int ret = avcodec_receive_packet(ctx->avctx, &pkt);
+			pkt = av_packet_alloc();
+			int ret = avcodec_receive_packet(ctx->avctx, pkt);
 			if (ret == AVERROR(EAGAIN)) {
 				break;
```
Comment (kennylevinsen) [inline examples/dmabuf-capture.c:497]: We still leak from here as ctx->err is not set (assuming that the write below here doesn't fail for zero-sized packets).

Comment (emersion) [reply examples/dmabuf-capture.c:497]: Eh, indeed. I've just reverted to free'ing the packet inside each branch.

**Review by kennylevinsen — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**